### ISSUE TITLE
Possible False Positive In button-name Rule

### ISSUE THREAD

**Issue by tmcconechy** [dequelabs/axe-core#831]
We are using icon buttons as follows

```
   <button type="button" class="btn-icon">
      <span>Date</span>
      <svg role="presentation" aria-hidden="true" focusable="false" class="icon">
        <use xlink:href="#icon-calendar"></use>
      </svg>
    </button>
```

This seems valid and does read the text in the span with the screen reader (its off screen so the button just looks like an icon. However, this rule:

https://dequeuniversity.com/rules/axe/3.0/button-name

Is complaining about it suggesting the use of aria-label.

Shouldn't the scanner check the internal text on spans when determining if the text is there on buttons? 

Also what if the span inside contains audible text like `<span class="audible">Date</span>`

**Comment by dylanb** [comment]
What is the CSS you are using for the `<span>Date</span>`?

**Comment by tmcconechy** [comment]
It has `visibility: hidden` changed it to `text-indent` and it now works as expected.
My mistake, this was a good pickup by your tool. I think a developer added this.

Sorry for that

**Comment by dylanb** [comment]
no problem...spread the word ;-)
